### PR TITLE
Dact 736 error order priority

### PR DESCRIPTION
**Pull request by mdesaikns**

### CODE REVIEW

File: src/controllers/director.residential.address.search.controller.ts [hunk 1]
```diff
@@ -51,7 +51,10 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
     const session: Session = req.session as Session;
     const postalCode : string = (req.body[DirectorField.POSTCODE])?.trim();
     const premise : string = (req.body[DirectorField.PREMISES])?.trim();
-    const jsValidationErrors = validatePremiseAndPostcode(postalCode, PostcodeValidation, PremiseValidation, premise);
+    let jsValidationErrors = validatePostcode(postalCode, PostcodeValidation);
+    if(premise) {
+      jsValidationErrors = validatePremise(premise, PremiseValidation, jsValidationErrors);
```
Comment (jbestwickch) [inline src/controllers/director.residential.address.search.controller.ts:56]: Won't this cause errors raised for the postcode to be lost? For character allow list validation I can imagine scenarios where we could get an error with both fields as an example

Comment (mdesaikns) [reply src/controllers/director.residential.address.search.controller.ts:56]: validatePremise() take an array if validationErrors as an argument - which will be populated by postcode validation if found any, and validatePremise() will return same array back with new errors if found in premise validation.
Screen Shot 2023-10-12 at 13 37 24 pm

Comment (jbestwickch) [reply src/controllers/director.residential.address.search.controller.ts:56]: I see, I don't think you need to reassign the variable in that case as javascript method parameters pass a ref to the object. Internal changes made will be reflected.

Comment (mdesaikns) [reply src/controllers/director.residential.address.search.controller.ts:56]: Thanks Jack, I wasn't aware of this. you're right. JS does pass a reference to an object.  
However I think it is generally a good practice to assign a variable when passing a reference to an object in a method,  this way makes code more readable easier to understand?

Comment (jbestwickch) [reply src/controllers/director.residential.address.search.controller.ts:56]: Might come down to personal preference/ coding style I guess. Not sure there's any guidance so happy to approve

**Review by jbestwickch — APPROVED** [review]

File: src/controllers/director.residential.address.search.controller.ts (hunk 1 above)
Comment (mdesaikns) [reply src/controllers/director.residential.address.search.controller.ts:56]: thanks Jack, I will wait for @companieshouse/team-thundercats to review and see if they have any specific guidance in such a case.

**Pull request merged**